### ISSUE TITLE
FR: About the Android taildrop notification problem

### ISSUE THREAD

**Issue by JarynChen** [tailscale/tailscale#8626]
### What are you trying to do?

I try to use taildrop to send files from other clients to Android. According to the official documentation, the Android side should receive a notification, click the notification to process the file, but I am using a mobile phone made in China and cannot receive the push from Google gcm. So my mobile phone can't receive the notification of file acceptance, so is there a better way to receive files without relying on message push

### How should we solve this?

My initial thought, can you open the api to handle the acceptance of files, such as adding a notification of file sending in webhook, I can build a push service to push messages to my mobile phone

### What is the impact of not solving this?

If there is no relevant optimization, Android phones made in China will not be able to use this function normally

**Comment by DentonGentry** [comment]
So the actual problem is that you don't get a notification when a taildrop file is delivered?

> but I am using a mobile phone made in China and cannot receive the push from Google gcm.

The notification is generated on the device, GCM is never used.

**Comment by JarynChen** [comment]
> So the actual problem is that you don't get a notification when a taildrop file is delivered?
> 
> > but I am using a mobile phone made in China and cannot receive the push from Google gcm.
> 
> The notification is generated on the device, GCM is never used.

Yes, the actual problem is that I did not receive relevant notifications on Android. At first I thought the notifications were through the gcm channel. I am using the miui14 Chinese mainland version. What information can I provide to help troubleshoot?

**Comment by jstebbins** [comment]
I also do not receive notifications when a file is delivered on android. The file shows up in downloads, but there is no notification.

Phone: Samsung S21 Ultra
Android 13

**Comment by mattlodder** [comment]
> I also do not receive notifications when a file is delivered on android. The file shows up in downloads, but there is no notification.
> 
> Phone: Samsung S21 Ultra Android 13

Same issue on a Pixel 6 Pro -- and Android tells me I can't turn on notifications for Tailscale as "This App Does not Send Notifications"...

**Comment by Jon-5** [comment]
I also do not receive no notification presented when files are received through Taildrop.

TailScale Version: 1.58.2
Install: Google Play Store

Phone: Pixel 7 Pro
Android version 14, build UQ1A.240205.002

I do remember in ~Dec 2022 seeing a notification upon receiving a file through Taildrop.

When querying the Android 'App info' screen it shows "Notifications are turned off" >tap> "This app does not send notifications", further "No Permissions granted" nor asked for by the app.

When Tailscale is enabled/disabled the VPN status bar icon does show/hide.

**Comment by attorneyatlaw** [comment]
I have two phone, one is galaxy note 20(SAMSUNG), another is G7(LG).
Galaxy do not receive notification, but G7 receives.

**Comment by afzl-wtu** [comment]
On Poco x6 pro (Android 14). Not working. Also I can not enable Notifications from App info settings

**Comment by ilyagr** [comment]
Same problem for me. This is in the US, not China, so I think Chinese issues may be a red herring. I'm using Pixel 6a, Android 14.

**Comment by phillipprado** [comment]
Can confirm this is still an issue. Tested on my Pixel 8 running Graphene OS, and the Tailscale app is installed from the Play Store. I tried multiple times, I tweaked the notifications settings, and I tried receiving files from both desktop and mobile devices. The file nearly instantly shows up in my downloads, but there is no notification when receiving it.

**Comment by yqs112358** [comment]
It seems that Tailscale Android use `direct mode` in ipnlocal to write the file received directly into Download, no any interaction with Android layer. It's an intentional act.

- https://github.com/tailscale/tailscale-android/blob/08ae01846890ba49702cee187cb1cb3fd0fbb5e0/android/src/main/java/com/tailscale/ipn/App.kt#L129
- https://github.com/tailscale/tailscale/blob/9248290941fc6d05aad96e7ef3a78ebeb3bdfca4/ipn/ipnlocal/local.go#L279

However, I think the pop up notification when a file is received is a necessary - this will allow the user to click and quickly open the received file, instead of opening file-manager manually and then searching through the whole file list.

Maybe something here https://github.com/tailscale/tailscale/blob/9248290941fc6d05aad96e7ef3a78ebeb3bdfca4/taildrop/taildrop.go#L101 can be useful to call callback functions registered by Kotlin, to provide features like progress-tracking and file receive notification.

**Comment by davidneedham** [comment]
I'm also seeing this. Google Pixel 6a. Android 14. 

I'm able to send files via Taildrop from Android to OSX, but it seems unreliable in the opposite direction. Two files did eventually show up on my Android device (no notification) but some are missing even though the share dialog on mac said that it sent.

**Comment by EDIflyer** [comment]
Same issue on Google Pixel 8 - no notification of file receipt and I can't find it on my device either, but the send seems to complete successfully.  No issue sending from Android to Windows or Linux, but can't seem to receive anything the other way round.

**Comment by njfife** [comment]
I don't get a notification on my pixel 9 pro fold, I tried to enable notifications and even restarted the device. 

I get the files in internal storage but no notification of any kind. I can work with that but it would still be nice to get a ping

**Comment by YueMiyuki** [comment]
Same problem on Xiaomi 11Ultra and Redmi K60 Pro. Both on stock ROM

**Comment by some0guy0** [comment]
Ditto for me on my Samsung Galaxy s24

**Comment by harry15111988** [comment]
I can confirm this same thing happened to me on Samsung S22 as well. Strangely, the first file did show up in the Gallery inside the Download folder fine. But the subsequent files only show when I navigate to `Internal Storage/Download/`

**Comment by clari7744** [comment]
A little shocked that this is still an issue. On my iPad, I've had issues getting the files to show up _unless_ I go through the notification, but the fact that no notification shows up at all on Android devices is really confusing. At least they go right to `~/Downloads` instead of a private storage, but I wasn't sure where to look at first.

**Comment by N8-Yue** [comment]
The issue is still a thing. No notification when transferring files through Taildrop

**Comment by drm-nz** [comment]
This was originally reported more than 2 years ago and it's still an issue on Android. Can someone please take a look at this?

**Comment by bruglet** [comment]
Would like to chime in and say that I still encounter this on my Pixel 8. If I pick a subfolder of Downloads as the destination, the file appears instantly and doesn't have any delay or reboots needed to see it, which was a former problem.

But the notifications still don't appear.

**Comment by MrSom3body** [comment]
It looks like this problem affects almost every Android device, as my phones and those of friends and family don't receive notifications

**Comment by iGesora** [comment]
Same issue using Android 16 on s23 Ultra and 14 on s21 Ultra, both have latest Tailscale. I can see a notification banner on the App icon but have to go into files>downloads to find the files and still banner shows. No notification on system tray, verified all Tailscale notifications are allowed.

**Comment by npeippo** [comment]
Problem still exists. Files not sending from Macos to Galaxy Fold 7. Other way it works fine.

**Comment by LukeHartell** [comment]
+1

Galaxy S23+
Android 16, One UI 8.5

**Comment by Rosaldivo** [comment]
Same here.
